### DISCUSSION TITLE
Why not allow using overlay control bar in window mode too?

### DISCUSSION THREAD

**Discussion by HubKing** [Rafostar/clapper#351] [Ideas]
In full-screen mode, it shows overlay control bar that appears when the mouse is over and automatically disappears after some time. So, the feature is already there.

 But in window mode, that is not available and either the control bar is docked at the bottom of the video, or not usable at all (when hidden by clicking the second button on the top-left, I could not find a way to access the bar). This is inconvenient. If the control bar is docked, it takes more space of my limited laptop monitor, but if it is not docked, I cannot use the control bar.

I could not find an option to make the overlay control bar work in window mode. Why not add an option for that? The feature is already there, so I do not see what the harm is.

**Comment by Rafostar** [comment]
Your Majesty,

One of the main reasons why controls panel was made beneath the video when windowed (thus relatively small) was not to cover too much of video itself like in the HIG designs of other GTK4 apps (interactive example [here](https://www.figma.com/proto/xzRerALc10Xf4Gr2rRfbMH/Polyphony-Replay?node-id=386%3A1998&viewport=572%2C333%2C0.3242415487766266&scaling=min-zoom&page-id=386%3A1997&starting-point-node-id=228%3A1649)). In such case most of video is covered, especially subtitles area.

Another reason was to distinguish if app is in always on top floating mode easily (if it does not have bottom controls, then it is). Additionally a bar that is constantly visible might be a plus if you want to quickly check (or observe) video time/position.

> or not usable at all

This mode is meant for specific use-case (viewing video in small window that stays on top of other windows on screen edge, something like PiP). For this purpose accidental hovering window with cursor that brings controls back would be more of annoyance. The video can still be played/paused in this mode with right click button for convenience.

> If the control bar is docked, it takes more space of my limited laptop monitor

The bar is not THAT high to take a considerable amount of space and if it is then watching videos fullscreen is always best.

**Reply by HubKing** [reply]
Well, first of all, the floating control bar on the video wouldn't not obscure the video too much, and the control bar is supposed to only appear when the mouse is over and automatically disappear when the mouse is not moving, so whatever obscured part of the video will be recovered in several seconds. 

When I was using Windows, PotPlayer was my video player and it had that auto-hiding UI mode, and I never thought the automatically appearing floating control was an obstacle; indeed I needed it. I used to play a video in PotPlayer as a small always-on-top window at the corner of my laptop screen whilst I was using other applications. I needed the floating control bar when I need to check the current video position or seek to a position. The case that you mentioned that I accidentally hover the mouse over the video window causing the floating control bar to appear and obscure the video barely happened or I don't remember feeling any inconvenience from that. 

After switching to Linux, I have been using MPV (vanilla so-called "pseudo UI") and MPV also acted like PotPlayer, showing the control bar over the video only when the mouse was moving over the video.

PotPlayer and MPV are both very popular video players, I believe, and if they work like that, doesn't that mean that there are lot of users who would want behaviours like that? I can understand maybe some users don't want floating control bar in window mode, but if that is turned off by default, and those users who want PotPlayer/MPV like behaviour can optionally enable it in the settings, and the feature is already there (floating control bar appears in full-screen mode of Clapper), what is the harm to let some users use that feature in window mode?

In case, if you say "If you like MPV that much, why not just use MPV instead of Clapper?", well, it is a "pseudo UI", and the developers seem to have Gnome, and I use Gnome, so it has its own various UI problems....
